Reorganize update scripts

gassmoeller opened the pull request:
This will make #2157 and #2163 simpler to merge, and also improves the organization of the update scripts. Currently, the update script is one sed-script that tries to update everything. Unfortunately, sed was a bad choice by me for this type of operation, because it has only one hold space, and its syntax is very hard to understand. With the new organization (one shell script in doc/ that calls all existing scripts in a subfolder), it will be simpler to add future changes in whatever language we prefer, and it prevents conflicts between different renaming operations. I also added more documentation to the manual, and made the application of the update script safer, by always generating backup files with the ending .bak. I also added a changelog entry.

jdannberg commented on doc/manual/manual.tex:
```diff
@@ -4061,6 +4061,26 @@ \subsubsection{A note on the syntax of formulas in input files}
+\texttt{doc/update\_prm\_files.sh} with one or more parameter files as
+arguments will create a backup of the old parameter file (named
+\texttt{old\_filename.bak}), and replace the existing file with a version that
+should work with the current \aspect{} version.
```
Oh, and can you also add a line for an example how you would call the update script? Some people might not be that familiar wish bash scripts, so let's make it easy for them.

jdannberg commented on doc/update_prm_files.sh:
```diff
@@ -0,0 +1,34 @@
+# we need to provide a backup file-ending (.tmp) and then later
+# remove the backup file. We can not use this file
+# instead of the .bak file, because it is overwritten by every script,
+# and so it is only a backup of the last execution.
```
Oh, so it's the same command for running the script for both linux and mac now? That's also a nice improvement!

gassmoeller commented:
I addressed your comments and added a new subsection for plugin files in the suggested section of the manual. Tests seem to be passing.

The script should now indeed be system independent, but I only implemented something I found on stackoverflow, maybe someone with a Mac could check?
@tjhei: Could you run `bash doc/update_prm_files.sh tests/update_script.prm` and check that it correctly changes update_script.prm in place, and creates an update_script.prm.bak with the old settings?

jdannberg commented:
All right, looks good, thank you!
If @tjhei (or someone else) can confirm this works works on a Mac, this can be merged.

naliboff commented:
@gassmoeller I got the following output from running `bash doc/update_prm_files.sh tests/update_script.prm` on my mac:
```
sed: 33: doc/update_scripts/prm_files/update_prm_files_to_2.0.0.sed: extra characters after \ at the end of i command
rm: tests/update_script.prm.tmp: No such file or directory
```

The file `/tests/update_script.prm.bak` was indeed created.

naliboff commented:
Works now 👍

gassmoeller commented:
Ok, thanks for the feedback.
